### PR TITLE
external/nanopb: Import nanopb

### PR DESCRIPTION
**Pull request by junmin-kim**
Add nanopb as a submodule

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [0b61bdfe5e69fd0937865f2b407c542dc71fd9a4] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0b61bdfe5e69fd0937865f2b407c542dc71fd9a4] - Code Rule Check OK.

File: external/nanopb/Kconfig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#
+# For a description of the syntax of this configuration file,
+# see kconfig-language at https://www.kernel.org/doc/Documentation/kbuild/kconfig-language.txt
+#
+
+config NANOPB
+	bool "Nanopb"
+	default n
+	---help---
+		enable Nanopb
```
Comment (Taejun-Kwon) [inline external/nanopb/Kconfig]: Could you explain about Nonopb little more here?

**Comment by seinfra** [comment]
Target : [71f9a94a567951af5191358bc470a4aefdb790ae] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [71f9a94a567951af5191358bc470a4aefdb790ae] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [6354ba7567083819dd89f663926275b77a7f8f9b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [6354ba7567083819dd89f663926275b77a7f8f9b] - Code Rule Check Result:
apps/examples/nanopb_simple/nanopb_simple_main.c:29: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
apps/examples/nanopb_simple/nanopb_simple_main.c:30: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
apps/examples/nanopb_simple/nanopb_simple_main.c:31: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
apps/examples/nanopb_simple/nanopb_simple_main.c:69: ERROR: [SPC_M_OPR] need consistent spacing around '/' (ctx:VxW)

**Comment by seinfra** [comment]
Target : [a8ef7f742127325d012d3756418f78639c721c88] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a8ef7f742127325d012d3756418f78639c721c88] - Code Rule Check OK.

File: apps/examples/nanopb_simple/Make.defs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+# OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+# AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+############################################################################
+
+ifeq ($(CONFIG_EXAMPLES_NANOPB_SIMPLE),y)
+CONFIGURED_APPS += examples/nanopb_simple
```
Comment (Taejun-Kwon) [inline apps/examples/nanopb_simple/Make.defs]: It is tightly related to CONFIG_NANOPB, please consider it. You may know what I mean.

**Comment by seinfra** [comment]
Target : [6db564f469663cbab275afe75c925d4e84217f76] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [6db564f469663cbab275afe75c925d4e84217f76] - Code Rule Check (C++) OK.

File: apps/examples/nanopb_simple/Make.defs (hunk 2 above)
Comment (junmin-kim) [reply apps/examples/nanopb_simple/Make.defs]: I fixed it.
Now CONFIG_EXAMPLES_NANOPB_SIMPLE depends on CONFIG_NANOPB
https://github.com/Samsung/TizenRT/pull/3963/files#diff-ca896b2d817f06e4220f6b4193c0f369R8

**Pull request merged**